internal/provider/resource_apply_module.go

```diff
@@ -300,16 +300,19 @@ func (t *applyModuleResource) Read(ctx context.Context,
 		return
 	}
 
-	// Update the state with the computed module source and version.
-	if currentApplied.moduleSource != nil {
-		state.ModuleSource = types.StringValue(*currentApplied.moduleSource)
-	} else {
-		state.ModuleSource = types.StringNull()
-	}
-	if currentApplied.moduleVersion != nil {
-		state.ModuleVersion = types.StringValue(*currentApplied.moduleVersion)
-	} else {
-		state.ModuleVersion = types.StringNull()
+	// If there is no current state version, currentApplied can be nil.
+	if currentApplied != nil {
+		// Update the state with the computed module source and version.
+		if currentApplied.moduleSource != nil {
+			state.ModuleSource = types.StringValue(*currentApplied.moduleSource)
+		} else {
+			state.ModuleSource = types.StringNull()
```

Comment:
Shouldn't the module source and module version be set to a null string value when the current applied is null?

Reply:
Implemented what I understood was the intent of this review item.